### PR TITLE
refactor: switch to pytest for sqlalchemy tests

### PR DESCRIPTION
**Pull request by juanrgon**
### Changes

- This PR changes our automated `sqlalchemy` tests to use [`pytest`](https://docs.pytest.org/en/7.1.x/) instead of [`nose`](https://nose.readthedocs.io/en/latest/)

### Why?

- I'm planning to add a `pytest` plugin to this library for query snapshot testing. This change will make this plugin testable

### Additional Minor Changes

- This PR updates our `setup.cfg` file to install `pytest` instead of `nose`. In addition, this PR updates the file to list all development and linting requirements on separate lines for better PR diffs in the future but to also provide better `git blame` reports

### Not in this PR

- No updates to the current tests in this PR. `pytest` supports non-`unittest`test cases, but this PR doesn't update our current tests to use this style. This will be done in a later PR since this will be needed to test the library's `pytest` plugin
- Not changing our Django tests to run with `pytest`. There is a [`pytest-django`](https://pytest-django.readthedocs.io/en/latest/) library which is supposed to add support for `pytest` functional testing to Django, but I have not tested this and it's out-of-scope for this PR

### What is the worse-case scenario for this PR, and why should we not worry about this?

**The tests could silently fail**

In theory, this PR could pass our github test actions but not actually fail. I don't think this is a legitimate concern, since locally I was seeing test failures running pytest when I did not have sqlalchemy installed.

**The tests could not actually be discovered**

In theory, `pytest` may be failing to find the existing `sqlalchemy` tests in this PR. The I don't think this is a legitimate concern, since the [`checks`](https://github.com/cedar-team/snapshot-queries/pull/21/checks) on this PR confirm that tests from the `tests/sqlalchemy` folder are running:

Screen Shot 2022-04-17 at 3 57 43 PM

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -22,8 +22,15 @@ install_requires =
     pygments>=2.8.1
```
Comment (juanrgon) [inline setup.cfg:22]: Decided to list each development and linting requirement in this file on separate lines to improve branch diffs and provider better git blames in the future

Comment (juanrgon) [inline setup.cfg:22]: Decided to list each development and linting requirement in this file on separate lines to improve branch diffs and provider better git blames in the future

**Review by ccrane11 — APPROVED** [review]

**Review by hack2learn — APPROVED** [review]
- [x] I fully understand what the code is doing and its context
- [x] The code is the best way to accomplish this problem (I would code it similarly)
- [x] The code is well organized, easy to understand and matches Cedar design patterns and best practices
- [x] The code handles edge cases (that I can think of) and has tests to validate them
- [x] The appropriate people are reviewing (knowledge and security)
- [ ] I tested the code personally (Not required)
